specs/FamilySafety.md

```diff
@@ -0,0 +1,103 @@
+}
+
+void FamilySafetyIframeFilterCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.IsFamilySafetyFirstLevelIFrameFilteringEnabled = !webView.IsFamilySafetyFirstLevelIFrameFilteringEnabled;
+    MessageBox.Show("Family Safety iframe filering is" + (webView.IsFamilySafetyFirstLevelIFrameFilteringEnabled ? " enabled " : " disabled ") + "after the next navigation.");
+}
+
+
+```
+## Win32 C++
+```cpp
+
+// Enable the Family Safety feature
+HRESULT ToggleFamilySafetyFeature()
+{
+    BOOL areFSEnabled;
+    wil::com_ptr<ICoreWebView2Staging3> webview3 =
+        m_webView.try_query<ICoreWebView2Staging3>();
+    CHECK_FAILURE(webview3->get_IsFamilySafetyFeatureEnabled(
+        &areFSEnabled));
+    CHECK_FAILURE(webview3->get_IsFamilySafetyFeatureEnabled(
+        !areFSEnabled));
+    MessageBox(
+        nullptr,
+        (std::wstring(L"Family Safety will be ") +
+            (!areFSEnabled ? L"enabled" : L"disabled") +
+            L" after the next restart.").c_str(),
+        L"Settings change", MB_OK);
+    return true;
+}
+
+// Toggle the iframe filter settings
+HRESULT ToggleFamilySafetyIframeFilterSettings()
+{
+    BOOL areIframeFilterEnabled;
+    wil::com_ptr<ICoreWebView2Staging3> webview3 =
+        m_webView.try_query<ICoreWebView2Staging3>();
+    CHECK_FAILURE(webview3->get_IsFamilySafetyFirstLevelIFrameFilteringEnabled(
+        &areIframeFilterEnabled));
+    CHECK_FAILURE(webview3->put_IsFamilySafetyFirstLevelIFrameFilteringEnabled(
+        !areIframeFilterEnabled));
+    MessageBox(
+        nullptr,
+        (std::wstring(L"Iframe Filtering will be ") +
+            (!areIframeFilterEnabled ? L"enabled" : L"disabled") +
+            L" after the next navigation.").c_str(),
+        L"Settings change", MB_OK);
+    return true;
+}
+```
+
+# API Details    
+```
+interface ICoreWebView2Staging3;
+
+/// A continuation of the ICoreWebView2 interface to toggle Family Safety settings
+[uuid(EF7E5FD3-6FAD-4065-8387-15A37288477E), object, pointer_default(unique)]
+interface ICoreWebView2Staging3 : IUnknown {
+  /// `IsFamilySafetyFeatureEnabled` property is to enable/disable family safety feature.
+  [propget] HRESULT IsFamilySafetyFeatureEnabled([out, retval] BOOL* value);
+
+  /// Sets the `IsFamilySafetyFeatureEnabled` property.
+  [propput] HRESULT IsFamilySafetyFeatureEnabled([in] BOOL value);
+
+  /// `IsFamilySafetyFirstLevelIFrameFilteringEnabled` property is to enable/disable first level iframe support in family safety.
+  [propget] HRESULT IsFamilySafetyFirstLevelIFrameFilteringEnabled([out, retval] BOOL* value);
+
+  /// Sets the `IsFamilySafetyFirstLevelIFrameFilteringEnabled` property.
+  [propput] HRESULT IsFamilySafetyFirstLevelIFrameFilteringEnabled([in] BOOL value);
+}
+```
+
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    // ...
+    runtimeclass CoreWebView2
```

Comment:
I am making some change to it. So rather than have live toggle. We only gonna toggle this upon each creation, and it should affect all instance in the same data folder. There is no profile support for fs.

Reply:
Ok so you're moving it to the ICoreWebView2EnvironmentOptions then?

Reply:
I did some testing and the setting applies to each new navigation. Does that mean it fit into setting interface better? But the  scenario is only for developer to toggle it and not expose to the user.

Reply:
When you say it applies to each new navigation, you mean its per CoreWebView2? Two CoreWebView2s in the same CoreWebView2Environment can have different family safety enabled values and they can both navigate at the same time and one will have family safety applied and the other won't? If so, having it on CoreWebView2Settings gives the end dev more flexibility to change it at runtime to handle scenarios we haven't thought of previously. But putting it on CoreWebView2EnvironmentOptions may make more sense for the end dev who is expecting to just turn it on once at the very beginning and not worry about it anymore. It also means they don't have to worry about any sort of race where the CoreWebView2 is performing some initial navigation but they haven't set the family safety property yet.

So, maybe putting it on the EnvironmentOptions makes more sense? I think its debatable and sounds like either way would work.